File: java/datastructure/src/test/java/otterop/datastructure/TestStringBuffer.java
package otterop.datastructure;

import otterop.lang.String;
import otterop.test.Test;
import otterop.test.TestBase;

public class TestStringBuffer extends TestBase {

    @Test
    public void empty() {
        StringBuffer sb = new StringBuffer();
        String s = sb.OOPString();
        this.assertTrue(s.compareTo(String.wrap("")) == 0,
                String.wrap("Should be an empty string"));
    }

    @Test
    public void addMoreStrings() {
        StringBuffer sb = new StringBuffer();
        sb.add(String.wrap("a"));
        String s = sb.OOPString();
        this.assertTrue(s.compareTo(String.wrap("a")) == 0,
                String.wrap("Should be equals to 'a'"));
        sb.add(String.wrap(",b"));
        s = sb.OOPString();
        this.assertTrue(s.compareTo(String.wrap("a,b")) == 0,
                String.wrap("Should be equals to 'a,b'"));
    }
}

File: dotnet/tests/Datastructure/Otterop/Datastructure/TestStringBuffer.cs
namespace Otterop.Datastructure
{
    public class TestStringBuffer : Otterop.Test.TestBase
    {
        [Otterop.Test.Test]
        public void Empty()
        {
            StringBuffer sb = new StringBuffer();
            Otterop.Lang.String s = sb.OOPString();
            AssertTrue(s.CompareTo(Otterop.Lang.String.Wrap("")) == 0, Otterop.Lang.String.Wrap("Should be an empty string"));
        }

        [Otterop.Test.Test]
        public void AddMoreStrings()
        {
            StringBuffer sb = new StringBuffer();
            sb.Add(Otterop.Lang.String.Wrap("a"));
            Otterop.Lang.String s = sb.OOPString();
            AssertTrue(s.CompareTo(Otterop.Lang.String.Wrap("a")) == 0, Otterop.Lang.String.Wrap("Should be equals to 'a'"));
            sb.Add(Otterop.Lang.String.Wrap(",b"));
            s = sb.OOPString();
            AssertTrue(s.CompareTo(Otterop.Lang.String.Wrap("a,b")) == 0, Otterop.Lang.String.Wrap("Should be equals to 'a,b'"));
        }

    }

}
